doc/api/stream.md

```diff
@@ -2132,9 +2135,15 @@ consistent and predictable handling of errors.
 const { Readable } = require('stream');
 
 const myReadable = new Readable({
+  autoDestroy,
   read(size) {
-    if (checkSomeErrorCondition()) {
-      process.nextTick(() => this.emit('error', err));
+    const err = checkSomeErrorCondition();
+    if (err) {
+      if (autoDestroy) {
+        this.destroy(err);
+      } else {
+        process.nextTick(() => this.emit('error', err));
```

Comment:
user code emitting error directly like this is typically not going to have the desired result. The user code should always just use `this.destroy(err)`

Reply:
> The user code should always just use this.destroy(err)

Then there is no way for user to emulated `autoDestroy`? I actually prefer your suggestion. Just making sure you are aware.